storage/watchable_store.go

```diff
@@ -268,9 +278,17 @@ func (s *watchableStore) syncWatchers() {
 	tx.Unlock()
 
 	for w, eb := range newWatcherBatch(&s.unsynced, evs) {
+		pr, ok := s.progresses[w.id]
+		if !ok {
+			panic(ErrWatcherNotExist)
+		}
+		status := WatchStatus{
```

Comment:
I still think this would be cleaner by just reusing `CompactRevision` with `Canceled = false`. Is there a reason why that wouldn't work here?